### PR TITLE
Add MVP voice assist flow

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Adds an MVP voice assist setup flow.

Copy needs to be updated (and translated if final)

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Release Notes

- **New Features**
  - Introduced a voice assistant setup dialog for streamlined configuration.
  - Added functionality for managing wake words and voice assistant pipelines.
  - Users can now select areas and wake words during the setup process.
  - Enhanced device configuration page with a direct "Set up voice assistant" action.
  - New methods for intercepting wake words and testing assistive satellite connections.
  - Added a component for managing the update process of the voice assistant.

- **Bug Fixes**
  - Improved error handling and conditional rendering in voice assistant pipeline details.

- **Style**
  - New styles for voice assistant setup components to enhance user experience.

- **Documentation**
  - Updated interfaces to support new optional properties for better flexibility in voice assistant configurations.

### CODE REVIEW

**Review by piitaya — DISMISSED** [review]
I added some comments but it's not really blocking. We can improve that later 🙂

**Review by silamon — COMMENTED** [review]
There are some unfinished or uncorrect translations in this PR. I would polish a bit more.

**Comment by piitaya** [comment]
@silamon This will not be available for beta as no device implement this. It will be at first for internal tests using nightly build.

File: src/dialogs/voice-assistant-setup/voice-assistant-setup-step-addons.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+export class HaVoiceAssistantSetupStepAddons extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _showFirst = false;
+
+  @state() private _showSecond = false;
+
+  @state() private _showThird = false;
+
+  @state() private _showFourth = false;
```
[the remark this answers is not in the document]
Comment (bramkragten) [reply src/dialogs/voice-assistant-setup/voice-assistant-setup-step-addons.ts:19]: Yeah but it gonna be removed anyway I think 🤷🏻‍♂️

**Pull request merged**